### DISCUSSION TITLE
Unattend (scripted) Installation of Homebrew

### DISCUSSION THREAD

**Discussion by mengelsen** [Homebrew/discussions#5463] [Getting started]
### Output of `brew config`

```shell
zsh: command not found: brew
```

### Output of `brew doctor`

```shell
zsh: command not found: brew
```

### Description of issue

According to the [installation documentation](https://github.com/Homebrew/install/#install-homebrew-on-macos-or-linux), I should be able to use the following command to perform a scripted install of Homebrew (on macOS):
`NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`
 
However, when I run this command, I get the following output:
```
==> Running in non-interactive mode because `$NONINTERACTIVE` is set.
==> Checking for `sudo` access (which may request your password)...
Need sudo access on macOS (e.g. the user it needs to be an Administrator)!
```
 
 
When I run the command as `sudo`, I get a different error:
```
sudo NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
Password:
==> Running in non-interactive mode because `$NONINTERACTIVE` is set.
==> Checking for `sudo` access (which may request your password)...
Don't run this as root!
```
 
 
How should I proceed with the unattend install of homebrew?

**Comment by Bo98** [comment]
`SUDO_ASKPASS` will allow you to pass in a password non-interactively to sudo.

**Reply by mengelsen** [reply]
I can give that a try.
But when I run the command without `SUDO_ASKPASS` I am prompted for a password. I supply the admin password (manually). Then, the install.sh script stops and displays "Don't run this as root!" on the screen.

So...if I include the `SUDO_ASKPASS` I'm thinking I'll get the same result.

**Reply by Bo98** [reply]
The Homebrew installer uses `sudo` internally and `NONINTERACTIVE` disables the sudo password prompt as it wouldn't be non-interactive otherwise. If you are deploying it in an environment that requires sudo passwords, then you will need to have some mechanism to non-interactively pass that password to sudo and `SUDO_ASKPASS` is a standard way to achieve that.

Although you shouldn't run the entire installer as sudo, the installer itself still uses sudo internally and will pass on `SUDO_ASKPASS` to it.

**Reply by bartvdbraak** [reply]
How does `SUDO_ASKPASS` work? I haven't found any documentation about it?

**Reply by SMillerDev** [reply]
https://www.man7.org/linux/man-pages/man8/sudo.8.html#ENVIRONMENT
